python/cuspatial/cuspatial/core/binpreds/binpreds.py

```diff
@@ -279,7 +162,28 @@ def preprocess(self, op, lhs, rhs):
         ).points
         return (lhs, final_rhs, point_indices)
 
-    def postprocess(self, op, point_indices, point_result):
+    def _op(self, lhs, points):
+        """Compute the contains_properly relationship between two GeoSeries.
+        A feature A contains another feature B if no points of B lie in the
+        exterior of A, and at least one point of the interior of B lies in the
+        interior of A. This is the inverse of `within`."""
+        if not contains_only_polygons(lhs):
+            raise TypeError(
+                "`.contains` can only be called with polygon series."
```

Comment:
I'm interested in this use case - I definitely want to support 1-to-many operations in the future. For now just using `NotImplementedError` is a good idea.

Reply:
Ok, I'll need to think about this. At this level, the `lhs` has to be polygon(s) because I'm about to call `lhs.polygons...` and the lhs must contain polygons. `.contains/.contains_properly` will be implemented for the other types as well, but they will be shortcutted during `preprocess` to another Binpred that doesn't require point-in-polygon for its result.

This involves part of the refactoring decision in the strategy pattern. The binpreds need to be able to call each other at different stages of other `preprocess, _op, or postprocess`. I haven't worked out how to design that in software most simply yet.

For now, `TypeError` here remains appropriate because in order to each this point the `lhs` must be polygon(s). Ok?

Reply:
Actually I'll move the same check up a level higher because I understand that this suggests that maybe the other combinations will not be implemented. They are not implemented yet because they do not depend on point-in-polygon.

Reply:
When intersect is merged, my plan is to iterate over each of the binary predicates, filling out the types matrix for each one. I'll create new documentation that details progress through each.

Reply:
From offline: let's add a docstring to `BinaryPredicate` to allow user to track the supported geometries.